math/src/field/fields/montgomery_backed_prime_fields.rs

```diff
@@ -128,7 +128,46 @@ where
         if a == &Self::ZERO {
             panic!("Division by zero error.")
         }
-        Self::pow(a, M::MODULUS - Self::BaseType::from_u64(2))
+        let mut u = M::MODULUS;
+        let mut v = a.clone();
+        let mut r = Self::zero();
+        let mut s = Self::one();
+        let mut k = 0;
+        let n = M::MODULUS.num_bits();
+
+        while v > Self::zero() {
```

Comment:
It's __typically__ faster to compare for equality as it should be bit comparison only, while in Montgomery it's not necessarily trivial to compute inequalities (as the comment below implies). Since `v` is unsigned, the only case where `v > 0` is false is when `v == zero`.
```suggestion
        while v != Self::zero() {
```